pkg/controller/virtualnode.go

```diff
@@ -264,12 +264,7 @@ func vnodeNeedsUpdate(desired *appmeshv1beta1.VirtualNode, target *aws.VirtualNo
 
 	// This needs to be updated since AppMesh VN name isn't the same as k8s VN name.
 	if desired.Spec.Backends != nil {
-		desiredSet := set.NewSet()
-		for i := range desired.Spec.Backends {
-			desiredSet.Add(desired.Spec.Backends[i])
-		}
-		currSet := target.BackendsSet()
-		if !desiredSet.Equal(currSet) {
+		if !reflect.DeepEqual(desired.Spec.Backends, target.Backends()) {
```

Comment:
will the AppMesh api *guarantee* the backends is ordered using same order when it's created with? guess that's the original intention to use set compare

Reply:
I suspect it is stable since it's a paginated API, but I don't think there's a guarantee. That makes sense. I don't recall why I removed this (maybe I thought it was redundant), but I'll add it back.

Reply:
I think what confused me is that `Listeners` is not handled the same way, though that might be because we have a limit of 1 from the api right now.